### PR TITLE
=str #16577 emit user exception from Flexi* completion handler

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

**Comment by ktoso** [comment]
LGTM

File: akka-stream/src/main/scala/akka/stream/impl/FlexiMergeImpl.scala [hunk 1]
```diff
@@ -44,7 +44,10 @@ private[akka] class FlexiMergeImpl(_settings: MaterializerSettings,
 
   override protected val inputBunch = new FanIn.InputBunch(inputPorts, settings.maxInputBufferSize, this) {
     override def onError(input: Int, e: Throwable): Unit = {
-      changeBehavior(completion.onError(ctx, inputMapping(input), e))
+      changeBehavior(try completion.onError(ctx, inputMapping(input), e)
+      catch {
+        case NonFatal(e) ⇒ fail(e); mergeLogic.SameState
```
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/FlexiMergeImpl.scala]: can this lead to double onError signals?

**Comment by drewhk** [comment]
LGTM

File: akka-stream/src/main/scala/akka/stream/impl/FlexiMergeImpl.scala (hunk 1 above)
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/FlexiMergeImpl.scala]: no, inputBunch and primaryOutputs have protection flag for that, so it is idempotent

**Comment by patriknw** [comment]
Refs #16577

**Pull request merged**